### PR TITLE
treewide: switch to fetchFromGitHub

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
use `fetchFromGitHub`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by jonringer — APPROVED** [review]
diff LGTM

```
https://github.com/NixOS/nixpkgs/pull/153282

1 package marked as broken and skipped:
natron

1 package blacklisted:
tests.nixos-functions.nixosTest-test

38 packages built:
atheme audacious axoloti badvpn clearlyU creep efont-unicode envypn-font fmbt gjay gohufont grub2 grub2_efi grub2_pvgrub_image grub2_xen hylafaxplus laminar libfaketime libmowgli libmx nxengine-evo os-prober pysideApiextractor pysideGeneratorrunner python38Packages.pyside python38Packages.pysideShiboken python38Packages.pysideTools python39Packages.pyside python39Packages.pysideShiboken python39Packages.pysideTools siji tests.trivial-builders.references tewi-font ucs-fonts uni-vga unifont woeusb woeusb-ng
```


Curious if we should target staging, just to save nixos infrastructure some compute and storage though. End users shouldn't notice these changes

**Comment by Stunkymonkey** [comment]
because `grub` gets rebuild?
for me that's fine. I just thought, that the amount of work is less for everybody if it is merged into master. Thats why I do multiple PRs

**Comment by Stunkymonkey** [comment]
Result of `nixpkgs-review pr 153282` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>37 packages built:</summary>
  <ul>
    <li>atheme</li>
    <li>audacious</li>
    <li>axoloti</li>
    <li>badvpn</li>
    <li>clearlyU</li>
    <li>creep</li>
    <li>efont-unicode</li>
    <li>envypn-font</li>
    <li>fmbt</li>
    <li>gjay</li>
    <li>gohufont</li>
    <li>grub2 (grub2_light)</li>
    <li>grub2_efi</li>
    <li>grub2_pvgrub_image</li>
    <li>grub2_xen</li>
    <li>hylafaxplus</li>
    <li>laminar</li>
    <li>libfaketime</li>
    <li>libmowgli</li>
    <li>libmx</li>
    <li>nxengine-evo</li>
    <li>os-prober</li>
    <li>pysideApiextractor</li>
    <li>pysideGeneratorrunner</li>
    <li>python38Packages.pyside</li>
    <li>python38Packages.pysideShiboken</li>
    <li>python38Packages.pysideTools</li>
    <li>python39Packages.pyside</li>
    <li>python39Packages.pysideShiboken</li>
    <li>python39Packages.pysideTools</li>
    <li>siji</li>
    <li>tewi-font</li>
    <li>ucs-fonts</li>
    <li>uni-vga</li>
    <li>unifont</li>
    <li>woeusb</li>
    <li>woeusb-ng</li>
  </ul>
</details>

**Comment by mohe2015** [comment]
I could imagine this could cause merge conflicts if done using staging

File: pkgs/games/nxengine-evo/default.nix [hunk 1]
```diff
@@ -10,9 +11,11 @@
...
   pname = "nxengine-evo";
   version = "2.6.4";
-  src = fetchurl {
-    url = "https://github.com/nxengine/nxengine-evo/archive/v${version}.tar.gz";
-    sha256 = "1xir74l0vrzrpyl7sfqfxp083dakbix26hd0arwf2y57w1mbjas8";
+  src = fetchFromGitHub {
+    owner = "nxengine";
+    repo = "nxengine-evo";
+    rev = "v${version}";
+    sha256 = "sha256-krK2b1E5JUMxRoEWmb3HZMNSIHfUUGXSpyb4/Zdp+5A=";
```
Comment (scubed2) [inline pkgs/games/nxengine-evo/default.nix:18]: Why does the hash change?

```
nix-hash --type sha256 --to-base16 1xir74l0vrzrpyl7sfqfxp083dakbix26hd0arwf2y57w1mbjas8
482bb96ae0a778e17856a041237a5c53b581c0ed0e3b7da8bff9e70d283939f6

nix-hash --type sha256 --to-base16 sha256-krK2b1E5JUMxRoEWmb3HZMNSIHfUUGXSpyb4/Zdp+5A=
92b2b66f513925433146811699bdc764c3522077d45065d2a726f8fd9769fb90
```

Comment (bobby285271) [reply pkgs/games/nxengine-evo/default.nix:18]: I think the reason for this is `fetchurl` stores a file exactly, `fetchFromGitHub` decompresses it.

Comment (scubed2) [reply pkgs/games/nxengine-evo/default.nix:18]: Ah, that makes sense. This looks good to me then.

**Review by scubed2 — APPROVED** [review]

**Review by bobby285271 — APPROVED** [review]
Checked all the hashes with nix-prefetch

**Pull request merged**